util/chplenv/chpl_launcher.py

```diff
@@ -63,10 +41,35 @@ def get():
             elif substrate_val == 'ofi':
                 launcher_val = slurm_prefix('gasnetrun_ofi')
         else:
-            if which('srun'):
-                launcher_val = 'slurm-srun'
+            is_known_cluster = (platform_val.startswith('cray-') or
+                                platform_val.startswith('hpe-'))
```

Comment:
Previous `starts_with` queries wouldn't evaluate to true for `cray-cs` or `hpe-apollo`. This one will. Is that a problem?

Even if not, this feels too wobbly to me (previous implementation was, too). Can we not have list of "known clusters" hard-coded in full instead of this loose name matching? I don't think that will be a very long list.

Reply:
> Previous starts_with queries wouldn't evaluate to true for cray-cs or hpe-apollo. This one will. Is that a problem?

I see that this part is intentional now.

The second part of my comment still stands, though.

Reply:
Sure, I can hard code the known platforms. Note this pattern of just checking a prefix is used all over chplenv for CHPL_[TARGET|HOST]_PLATFORM. I don't consider it that wobbly, but I also am not opposed to locking down a specific list for this case

Reply:
If the resulting inconsistency in the script makes things worse in your judgement, feel free to ignore this comment.

Reply:
Looking through the code again, I think making another helper similar to `is_hpe_cray` and `is_cray` is the best option. I'll implement that now